### PR TITLE
Introduce the meta_like function

### PR DESCRIPTION
**Pull request by cbalioglu**
This PR introduces a new convenience function `meta_like()` in fake tensor API to convert a fake tensor to a meta tensor that is more efficient than a call to `Tensor.to()`. It also has detach semantic meaning the returned tensor won't be part of the Autograd graph.

```
import torch
from torchdistx import fake

with fake.fake_mode():
    a = torch.ones([10])

b = fake.meta_like(a)

print(a)
print(b)
```

```
tensor(..., size=(10,), fake=True)
tensor(..., device='meta', size=(10,))
```

### CODE REVIEW

File: tests/python/test_fake.py [hunk 1]
```diff
@@ -4,6 +4,26 @@
...
+def test_meta_like_works_as_expected():
+    with fake_mode():
+        a = torch.ones([10])
+
+    b = meta_like(a)
+
+    assert b.device.type == "meta"
+    assert b.dtype == a.dtype
+    assert b.size() == a.size()
+    assert b.stride() == a.stride()
```
Comment (rohan-varma) [inline tests/python/test_fake.py:22]: Shall we test more complex cases such as non-contiguous or sparse tensors?

**Review by rohan-varma — DISMISSED** [review]
LGTM! Just 2 small questions

File: tests/python/test_fake.py [hunk 2]
```diff
@@ -4,6 +4,26 @@
...
-def test_foo():
-    assert True
+from torchdistx.fake import fake_mode, meta_like
+
+
+def test_meta_like_works_as_expected():
+    with fake_mode():
+        a = torch.ones([10])
+
+    b = meta_like(a)
```
[the remark this answers is not in the document]
Comment (cbalioglu) [reply tests/python/test_fake.py:17]: Right. Just pushed a revision that also assert that `b` is not fake.

File: tests/python/test_fake.py (hunk 1 above)
Comment (cbalioglu) [reply tests/python/test_fake.py:23]: There is no such thing as a sparse meta tensor (equivalently sparse fake tensor). A meta/fake tensor always represents a dense tensor. COO and CSR tensors in PyTorch are build as compositions of dense tensors but their meta backend coverage for sparse operators is pretty weak for now.

Comment (rohan-varma) [reply tests/python/test_fake.py:23]: I see, thanks for explaining!
